src/templates/TemplateData.ts

```diff
@@ -0,0 +1,105 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+// tslint:disable-next-line:no-require-imports
+import request = require('request-promise');
+import * as vscode from 'vscode';
+import { localize } from '../localize';
+import { Config } from './Config';
+import { ConfigBinding } from './ConfigBinding';
+import { ConfigSetting } from './ConfigSetting';
+import { Resources } from './Resources';
+import { Template, TemplateCategory, TemplateLanguage } from './Template';
+
+/**
+ * Main container for all template data retrieved from the Azure Functions Portal. See README.md for more info and example of the schema.
+ * We cache the template data retrieved from the portal so that the user can create functions offline.
+ */
+export class TemplateData {
+    private readonly _templateInitError: Error = new Error(localize('azFunc.TemplateInitError', 'Failed to retrieve templates from the Azure Functions Portal.'));
```

Comment:
Azure Portal

Reply:
There's a separate portal just for functions

---
In reply to: [147806673](https://github.com/Microsoft/vscode-azurefunctions/pull/32#discussion_r147806673) [](ancestors = 147806673)